cs/src/core/Allocator/AllocatorBase.cs

```diff
@@ -1334,29 +1356,7 @@ private void PageAlignedShiftReadOnlyAddress(long currentTailAddress)
         /// </summary>
         /// <param name="currentTailAddress"></param>
         private void PageAlignedShiftHeadAddress(long currentTailAddress)
-        {
-            //obtain local values of variables that can change
-            long currentHeadAddress = HeadAddress;
-            long currentFlushedUntilAddress = FlushedUntilAddress;
-            long pageAlignedTailAddress = currentTailAddress & ~PageSizeMask;
-            long desiredHeadAddress = (pageAlignedTailAddress - HeadOffsetLagAddress);
-
-            long newHeadAddress = desiredHeadAddress;
-            if (currentFlushedUntilAddress < newHeadAddress)
-            {
-                newHeadAddress = currentFlushedUntilAddress;
-            }
-            newHeadAddress &= ~PageSizeMask;
-
```

Comment:
newHeadAddress &= ~PageSizeMask;
is not in the new call-through version. It seems needed if wew set newHeadAddress = currentFlushedUntilAddress;

Reply:
I saw that earlier, but it is a benign change. The reason we have PageAligned is to avoid many small head movements as the tail grows, and this version satisfies that requirement.